supabase/models/messaging/newConvoStart.ts

```diff
@@ -8,13 +8,34 @@ export default async function newConvoStart(
   if (userID && donorID && itemID) {
     try {
       const supabase = newClient();
-      const { data: conversationData } = await supabase
+      /**
+       * @description before creating a new conversation, we check if such a conversation already exists in the database,
+       * if we find such a conversation, we return its ID that we can use further in the code,
+       * if such a conversation does not exist in the database, then we create a new conversation
+       */
+      const { data: existingConversations, error } = await supabase
```

Comment:
I love seeing docs added to PR's 😍 

I'd suggest this one get's moved up to the root function declaration (line 3) and gets a few more labels:

```ts
/**
 * 
 * @description checks if a conversation matching these parameters already exists before attempting to create a new one
 * @param userID Id of the user currently logged in
 * @param donorID Id of the user who is donating this item
 * @param itemID Id of the item which the new conversation will relate to
 * @returns null if no conversation existed previously or conversation_id if an existing conversation has been found
 */
export default async function newConvoStart(
  userID: string | undefined,
  donorID: string,
  itemID: string
) {
```

The reason is that this will create a more detailed label whenever we hover on `newConvoStart` elsewhere in the code and since you've now added the return of an Id I can see this function getting used for a few extra tweaks down the line so it will be useful to get a quick glance at it's return value